packages/react-core/src/components/Page/PageHeaderToolsItem.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Page/page';
+import { css } from '@patternfly/react-styles';
+import { PageHeaderToolsBreakpointMod } from './PageHeaderTools';
+import { formatBreakpointMods } from '../../helpers/util';
+
+interface InjectedPageHeaderToolsItemProps {
+  selectedClass: string;
+}
+
+export interface PageHeaderToolsItemProps extends React.HTMLProps<HTMLDivElement> {
+  /** Content rendered in page header tools item. If you need the selected class, use the render prop instead */
+  children?: React.ReactNode;
+  /** If you need the selected class, use this render prop instead of children */
+  render?: (props: InjectedPageHeaderToolsItemProps) => React.ReactNode;
```

Comment:
opened issue in core so this hopefully won't be necessary to get the selected class
https://github.com/patternfly/patternfly/issues/3108

Reply:
We haven't really used an API like this before, but I like it and I think it satisfies the documentation part.

Reply:
Yes, I remembered there was some controversy about using `cloneElement` so render prop puts the resposibility of adding the class back to the user. anyways, this render prop will be short-lived once the mentioned core PR is available :)